test/test-component.js

```diff
@@ -86,6 +86,41 @@ describe('Simple Component instance', function() {
       });
     });
   });
+
+  context('when using shadow DOM', function() {
+    beforeEach(function(done) {
+      el = document.createElement('shadow-dom-app');
+      document.body.appendChild(el);
+      window.requestAnimationFrame(function() {
+        done();
+      });
+    });
+
+    it('creates and uses a shadow root', function() {
+      expect(el.el).not.to.equal(el);
+      expect(el.shadowRoot).to.be.ok;
+    });
+
+    it('renders its template', function() {
+      expect(document.getElementsByClassName('foo')).to.have.lengthOf(0);
+      expect(el.children).to.have.lengthOf(0);
+      expect(el.shadowRoot.children[1].className).to.equal('foo');
+    });
+
+    it('applies the styles', function() {
+      expect(el.shadowRoot.children[0].innerHTML).to.equal('color: blue;');
+    });
+
+    context('when applying override styles', function() {
```

Comment:
can we have one test case where the override is already set before `attachedCallback` fires? e.g. `<my-widget style-override="background: red">`